### ISSUE TITLE
Legend Fails with BarGraphItem

### ISSUE THREAD

**Issue by YuMan-Tam** [pyqtgraph/pyqtgraph#1171]
The symbol with respect to a ```BarGraphItem``` does not show up in the legend.

The issue can be reproduced using the following code.

```
import pyqtgraph as pg
from pyqtgraph.Qt import QtCore, QtGui
import numpy as np

win = pg.plot()
win.setWindowTitle('pyqtgraph example: BarGraphItem')

x = np.arange(10)
y = np.sin(x+2) * 3
bg1 = pg.BarGraphItem(x=x, height=y, width=0.3, brush='b', pen='w', name='bar')
win.addItem(bg1)

legend = pg.LegendItem((80,60), offset=(70,20))
legend.setParentItem(win.graphicsItem())
legend.addItem(bg1, 'bar')
```
The code produces the following error:

```
File "/usr/local/lib/python3.7/site-packages/pyqtgraph/graphicsItems/LegendItem.py", line 237, in paint
    if opts['antialias']:
KeyError: 'antialias'
```
The error disappear after adding ```antialias = False``` to the ```BarGraphItem``` but this does not fix the issue I raised. I suspect this is related to the open issue https://github.com/pyqtgraph/pyqtgraph/pull/800.

My package version is ```0.11.0.dev0+g88931bc```. I wonder if this is a bug or there are some tricks that would get around this issue?

Thank you!

**Comment by 2xB** [comment]
@SamSchott You revisited the legend recently, do you have any insight on this, especially on the error?

**Comment by SamSchott** [comment]
Most graphics items have an antialias option, such as [PlotCurveItem](https://github.com/pyqtgraph/pyqtgraph/blob/develop/pyqtgraph/graphicsItems/PlotCurveItem.py#L69). Part of the revisited legend included drawing the symbol with antialiasing if this is done for the graphics item itself. However, `BarGraphItem` apparently does not have an antialias option - this was an oversight.

There are three options to fix this, with increasing effort:

1. Use `opts.get('antialias', False)` instead of `opts['antialias']` when drawing the legend symbol.
2. Ensure that all graphics items have an antialias option.
3. Let every graphics item decide itself how its legend symbol should look like.

Regarding the issue of legend symbol not showing: It does show for me, but only as a line (with the pen set for the `BarGraphItem` and the brush ignored). This is to be expected since there is currently no special handing of legend items for bar graphs.

**Comment by YuMan-Tam** [comment]
Thank you for your reply. It could be that I did not see the symbol because of the black background color. The pen set and brush were definitely not shown.

I believe the issue is settled for now. Should I close the issue or leave it open for the ```antialias``` bug and/or an feature request for the ```BarGraphitem```?

**Comment by 2xB** [comment]
@YuMan-Tam Since there is a remaining issue here, I would leave it open.

**Comment by j9ac9k** [comment]
Fixed with #1228
